Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.cpp

```diff
@@ -280,6 +394,22 @@ cairo_surface_t* webkit_favicon_database_get_favicon_finish(WebKitFaviconDatabas
 }
 #endif
 
+#if PLATFORM(WPE) && ENABLE(2022_GLIB_API)
+WebKitFavicon* webkit_favicon_database_get_favicon_finish(WebKitFaviconDatabase* database, GAsyncResult* result, GError** error)
+{
+    g_return_val_if_fail(WEBKIT_IS_FAVICON_DATABASE(database), nullptr);
+    g_return_val_if_fail(g_task_is_valid(result, database), nullptr);
+
+#if USE(SKIA)
+    auto image = sk_sp<SkImage>(static_cast<SkImage*>(g_task_propagate_pointer(G_TASK(result), error)));
+#else
+    auto image = adoptGRef(static_cast<cairo_surface_t*>(g_task_propagate_pointer(G_TASK(result), error)));
+#endif
+
+    return WebKitFavicon::create(WTFMove(image));
```

Comment:
AFAIU we will always get a RGBA8888 image here, and that's the reason why I have not added a `webkit_favicon_get_format()` getter.

If we might get something else here, we have two options: a.) always convert to a given format before creating the `WebKitFavicon`, and RGBA8888 should be fine for any kind of favicon IMO; or b.) add the getter for the pixel format and let client code deal with conversions and who knows how many possible formats. One last option could be to do both: add `webkit_favicon_get_format()` _and_ a `webkit_favicon_as_rgba8888()` that converts if needed.

Thoughts welcome!

Reply:
Correction: we currently get BGRA8888 on little-endian.